app/views.py

```diff
@@ -0,0 +1,339 @@
+    validation = validate_bucketlist(json)
+    if validation.status:
+        bucketlist = BucketList(name=json['name'])
+        current_identity.bucketlists.append(bucketlist)
+        db.session.add(bucketlist)
+        db.session.commit()
+        status_code = 201
+    else:
+        status_code = 400
+
+    response["message"] = validation.message
+    response = jsonify(response)
+    response.status_code = status_code
+    return response
+
+
+@app.route("/bucketlists", methods=['GET'])
+@jwt_required()
+def get_all_bucketlists():
+    '''
+    List all the created bucket lists and search bucketlists by name
+    '''
+    response = {}
+    name = request.args.get('q')  # search string
+    limit = request.args.get('limit')
+    offset = request.args.get('offset')
+
+    if name is None:
+        validation = validate_limit_and_offset(limit, offset)
+        if validation.status:
+            response["bucketlists"] = current_identity.get_bucketlists_as_json(
+                limit, offset)
+            response["meta"] = {}
+            status_code = 200
+
+            if len(response["bucketlists"]) < 1:
+                status_code = 404
+                response["message"] = "No bucketlists exist."
+        else:
+            status_code = 400
+            response[
+                "message"] = "Invalid format for limit or offset. Should be integer"
+
+    elif name is not None and len(name) > 0:
+        response["bucketlists"] = []
+
+        name = name.lower()
+        query = "%" + name + "%"
+        bucketlists = BucketList.query.filter(
+            BucketList.name.like(query)).filter_by(
+            owner_id=current_identity.user_id).all()
+
+        if bucketlists:
+            for bucketlist in bucketlists:
+                response["bucketlists"].append(bucketlist.to_json())
+
+            status_code = 200
+
+        else:
+            status_code = 404
+            response["message"] = "No bucketlists by that name found."
+
+    else:
+        status_code = 400
+
+    response = jsonify(response)
+    response.status_code = status_code
+    return response
+
+
+@app.route("/bucketlists/<int:bucketlist_id>", methods=['GET'])
+@jwt_required()
+def get_single_bucketlist(bucketlist_id):
+    response = {}
+    response["bucketlist"] = {}
+    response["meta"] = {}
+
+    user_id = current_identity.user_id
+    bucketlist = BucketList.query.filter(
+        bucketlist_id == BucketList.bucketlist_id,
+        user_id == BucketList.owner_id
+    ).first()
+
+    if bucketlist is not None:
+        response["bucketlist"] = bucketlist.to_json()
+        status_code = 200
+    else:
+        status_code = 404
+        response["message"] = "The requested bucketlist does not exist."
+
+    response = jsonify(response)
+    response.status_code = status_code
+    return response
+
+
+@app.route("/bucketlists/<int:bucketlist_id>", methods=['PUT'])
+@jwt_required()
+def update_single_bucketlist(bucketlist_id):
```

Comment:
`def update_single_bucketlist(bucketlist_id):` could also be joined with `def get_single_bucketlist(bucketlist_id):` and `def delete_single_bucketlist(bucketlist_id):` using `@app.route("/bucketlists/<int:bucketlist_id>", methods=['GET', 'PUT', 'DELETE'])`